python/graphstorm/gconstruct/utils.py

```diff
@@ -42,6 +44,14 @@ def worker_fn(task_queue, res_queue, user_parser):
     user_parser : callable
         The user-defined function to read and process the data files.
     """
+    # We need to set a GPU device for each worker process in case that
+    # some transformations (e.g., computing BERT embeddings) require GPU computation.
+    if th.cuda.is_available():
+        num_gpus = th.cuda.device_count()
+        gpu = worker_id % num_gpus
+        os.environ['CUDA_VISIBLE_DEVICES'] = str(gpu)
+        if worker_id >= num_gpus:
```

Comment:
In each worker, can we check whether worker_id > num_gpus? If so, the worker just return an empty result.

Reply:
This can help avoid potential OOM problem.

Reply:
I give a warning here. we should allow the number of processes to be larger than the number of GPUs. some of the feature operations run on CPU. currently, only bert computations run on GPUs.

Reply:
this is not so good either. some feature transformations don't require GPUs. the message will be confusing.

Reply:
OK.
Are we sharing the Bert model between workers using the same GPU or each worker needs to load the BERT model?

I am fine with current implementation. But we would better optimize it later.

Reply:
the bert model is not shared. when computing bert embeddings, we should ask users to set the number of trainers correctly.

Reply:
another option is to detect whether the feature transformation requires GPU. if it does, we set the number of trainer processes to the number of available GPUs automatically.

Reply:
I would prefer the later one.